runtime/common/src/parachains.rs

```diff
@@ -426,106 +368,20 @@ impl<T: Trait> Module<T> {
 		Ok(())
 	}
 
-	/// Update routing information from the parachain heads. This queues upwards
-	/// messages to the relay chain as well.
-	fn update_routing(
-		now: T::BlockNumber,
-		heads: &[AttestedCandidate],
-	) {
-		// TODO: per-chain watermark
-		// https://github.com/paritytech/polkadot/issues/286
-		let watermark = now.saturating_sub(One::one());
-
-		let mut ingress_update = BTreeMap::new();
-
-		// we sort them in order to provide a fast lookup to ensure we can avoid duplicates in the
-		// needs_dispatch queue.
-		let mut ordered_needs_dispatch = NeedsDispatch::get();
-
-		for head in heads.iter() {
-			let id = head.parachain_index();
-			<Heads>::insert(id, &head.candidate.head_data.0);
-
-			let last_watermark = <Watermarks<T>>::mutate(id, |mark| {
-				rstd::mem::replace(mark, Some(watermark))
-			});
-
-			if let Some(last_watermark) = last_watermark {
-				// Discard routed ingress.
-				for routed_height in number_range(last_watermark, watermark) {
-					<UnroutedIngress<T>>::remove(&(routed_height, id));
-				}
-			}
-
-			// place our egress root to `to` into the ingress table for (now, `to`).
-			for &(to, root) in &head.candidate.egress_queue_roots {
-				ingress_update.entry(to).or_insert_with(Vec::new).push((id, root));
-			}
-
-			// Queue up upwards messages (from parachains to relay chain).
-			Self::queue_upward_messages(
-				id,
-				&head.candidate.upward_messages,
-				&mut ordered_needs_dispatch,
-			);
-		}
-
-		NeedsDispatch::put(ordered_needs_dispatch);
-
-		// apply the ingress update.
-		for (to, ingress_roots) in ingress_update {
-			<UnroutedIngress<T>>::insert((now, to), ingress_roots);
-		}
-	}
-
-	/// Place any new upward messages into our queue for later dispatch.
-	///
-	/// `ordered_needs_dispatch` is mutated to ensure it reflects the new value of
-	/// `RelayDispatchQueueSize`. It is up to the caller to guarantee that it gets written into
-	/// storage after this call.
-	fn queue_upward_messages(
```

Comment:
This should be reinstated as it's parachain -> relay-chain messaging, not parachain -> parachain